ADDING input flag for controlling ammount of hostpitals used

ivanBum opened the pull request:
changes input flag for the hospital import to select a Verily specific csv file.
Adds another input flag that indicates how many hospitals will it select from the list. It doesn't check if the imput number is valid.
Function only selects the first n rows of the hospitals csv and uses them for data generation.

ivanBum committed:
New Hospital File

ivanBum committed:
Only Massachusetts Hospitals

ivanBum committed:
Sandbox

ivanBum committed:
properties

ivanBum committed:
Set up functions & properties

ivanBum committed:
Generator

ivanBum committed:
ReducedCSV working function

ivanBum committed:
Only Massachusetts Hospitals

ivanBum committed:
Adding exceptions, restrictions and best practices

ivanBum commented:
10k lines are from a csv file

pratikchoudhary commented on sandbox/removeRows.py:
```diff
@@ -0,0 +1,15 @@
+import pandas as pd
+
+data = pd.read_csv(
+    "/usr/local/google/home/bumaschny/synthea/src/main/resources/providers/hospitals.csv"
```
Please use relative file path. 
for example: 
`../synthea/src/main/resources/providers/hospitals.csv`

pratikchoudhary commented on sandbox/removeRows.py:
```diff
@@ -0,0 +1,15 @@
+
+
+data.to_csv(
+    '/usr/local/google/home/bumaschny/synthea/src/main/resources/providers/hospitals_verily.csv')
```
same as above.

pratikchoudhary commented on sandbox/removeRows.py:
```diff
@@ -0,0 +1,15 @@
+    "/usr/local/google/home/bumaschny/synthea/src/main/resources/providers/hospitals.csv"
+)
+
+print(len(data.index))
```
do we need to print to stdout?
If we do, maybe append some description of what is to be printed at the beginning. For example:
`print(f"Num of rows: {len(data.index)}")`

pratikchoudhary commented on sandbox/removeRows.py:
```diff
@@ -0,0 +1,15 @@
+data.to_csv(
+    '/usr/local/google/home/bumaschny/synthea/src/main/resources/providers/hospitals_verily.csv')
+
+print(len(data.index))
```
Same as above.

pratikchoudhary commented on src/main/java/org/mitre/synthea/engine/Generator.java:
```diff
@@ -919,4 +933,59 @@ private Predicate<String> getModulePredicate() {
     return this.populationRandom;
   }
+
+  public void reduceNumberOfHospitals() {
```
Can you please add a java doc to explain what the method is doing.

ivanBum replied:
Added Javadoc

pratikchoudhary commented on src/main/java/org/mitre/synthea/engine/Generator.java:
```diff
@@ -919,4 +933,59 @@ private Predicate<String> getModulePredicate() {
+    // Verily CSV
+    String verilyFile = "src/main/resources/providers/hospitals_verily.csv";
+    File csvFile = new File(verilyFile);
+    if (!(csvFile.exists() && !csvFile.isDirectory())) {
```
should this just return? Why not throw an exception?

ivanBum replied:
Fixed it.
``` 
try {
      if (!(csvFile.exists() && !csvFile.isDirectory())) {
        System.out.println("ERROR: File " + verilyFile + "Does not exists.\n");
      }        
    } catch (Exception e) {}
```

pratikchoudhary commented on src/main/java/org/mitre/synthea/engine/Generator.java:
```diff
@@ -919,4 +933,59 @@ private Predicate<String> getModulePredicate() {
+    String verilyReducedFile = new File(hospitalFile).getAbsolutePath();
+    File reducedCSVFile = new File(verilyReducedFile);
+    if (!(reducedCSVFile.exists() && !reducedCSVFile.isDirectory())) {
+      System.out.println("ERROR: File " + verilyReducedFile + " Does not exists.\n");
```
Why should this print an error if this is expected?

ivanBum replied:
I'll change it to `Warning:` what do you think?

pratikchoudhary commented on src/main/java/org/mitre/synthea/engine/Generator.java:
```diff
@@ -919,4 +933,59 @@ private Predicate<String> getModulePredicate() {
+
+  public void reduceNumberOfHospitals() {
+    // +1 Since this includes the header
+    int hospitalNumber = Integer.parseInt(Config.get("verily.limit_hospital_number")) + 1;
```
why are we adding +1 to the limit? The limit is 4, but we set it to 5 here.
If we want the limit to be 5, why not just set that in the properties?

ivanBum replied:
Because by flag (synthea.properties) you'll want to limit the hospital number, to follow this example, to 4. Since you'll also want to load the header, that means you'll need a +1. 
If you take out the +1, you'll need to add it via input (synthea.properties = 5) and I think this way is more clear on the ammount of hospitals you want to limit it to

pratikchoudhary reviewed (commented):
Thanks for the PR Ivan!
I've added a few comments.
One common feedback, instead of printing the errors and keep on going, why don't we fail fast and throw an Exception with a message instead?

ivanBum committed:
Choose number of hospitals - constant names

ivanBum committed:
modified .gitignore

ivanBum committed:
Delete removeRows.py

File should've not been uploaded, including it to .gitignore

ivanBum committed:
Merge branch 'Verily' into reduce_hospitals

ivanBum committed:
Merge branch 'reduce_hospitals' into two_hospitals

ivanBum committed:
Delete hospitals_verily_reduced.csv

Run the program and get the error "No hospitals_verily_reduced.csv; creating one". Then run again

ivanBum committed:
Merge pull request #11 from ivanBum/two_hospitals

Two hospitals

ivanBum committed:
Javadoc

ivanBum commented:
Added `chooseHospitals()` function & answered comments. Please review this new method (it has a javadoc explaining what it does).
Also, if you can, please test it.

ivanBum committed:
Correcting for PR review

ivanBum committed:
PR Review changes

ivanBum committed:
Merge branch 'Verily' into reduce_hospitals

ivanBum committed:
Standard Config

ivanBum commented:
To use this functionality:
```
./run_synthea --verily.choose_hospitals true
```
In the `src/main/java/org/mitre/synthea/engine/Generator.java` class, go to the chooseHospitals function and change the `HOSPITAL_n` strings to those hospitals names you'll want to use.
To only reduce the nomber of hospitals without selecting them, leave the `verily.choose_hospitals` flag `false`and change the `verily.limit_hospital_number` flag inside `src/main/resources/synthea.properties` to the number of hospitals you want to use.

**IMPORTANT**:
You need to have a file called `hospitals_verily` under the `providers/` directory (which is included in this pr); this `hospitals` file differs from the original one by only having Massachusetts hospitals. 
In the case you'll want to use another place, you need to follow two steps:
1. change the  `hospitals_verily` csv to only have hospitals from the new place
2. change default population location to said place
